### PR TITLE
TimeVal: Do a runtime check for CLOCK_BOOTTIME support

### PR DESCRIPTION
**Pull request by NiLuJe**
And create a few aliases with an automatic fallback to another clock source.

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/8832)

### CODE REVIEW

**Comment by NiLuJe** [comment]
Re: #8815 ;).

File: frontend/ui/timeval.lua [hunk 1]
```diff
@@ -25,6 +25,8 @@ local C = ffi.C
 local PREFERRED_MONOTONIC_CLOCKID = C.CLOCK_MONOTONIC
 -- Ditto for REALTIME (for :realtime_coarse only, :realtime uses gettimeofday ;)).
 local PREFERRED_REALTIME_CLOCKID = C.CLOCK_REALTIME
+-- CLOCK_BOOTTIME is only available on Linux 2.6.30+...
```
[the remark this answers is not in the document]
Comment (NiLuJe) [reply frontend/ui/timeval.lua:28]: I may have dug into arm kernels to pull the version relevant for arm ;).

(or not, can't remember :D).

File: frontend/ui/timeval.lua [hunk 2]
```diff
@@ -205,6 +214,19 @@ function TimeVal:boottime()
...
 end
 
+--- Since CLOCK_BOOTIME may not be supported, we offer a few aliases with automatic fallbacks to MONOTONIC or REALTIME
+if HAVE_BOOTTIME then
+    TimeVal.boottime_or_monotonic = TimeVal.boottime
+    TimeVal.boottime_or_monotonic_coarse = TimeVal.boottime
+    TimeVal.boottime_or_realtime = TimeVal.boottime
+    TimeVal.boottime_or_realtime_coarse = TimeVal.boottime
+else
+    TimeVal.boottime_or_monotonic = TimeVal.monotonic
```
[the remark this answers is not in the document]
Comment (NiLuJe) [reply frontend/ui/timeval.lua:224]: That's a tough call. I wouldn't mind *adding* a msg, but it should still return a zero TV.

File: frontend/ui/timeval.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (NiLuJe) [reply frontend/ui/timeval.lua:28]: Nope, that makes no sense, since 30 < 39, arm usually comes *later*. Must have been a typo ;).

**Comment by zwim** [comment]
lgtm.

**Review by zwim — APPROVED** [review]

**Comment by zwim** [comment]
Ready to merge?

**Pull request merged**